sdk/go/x/auto/git.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2016-2020, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package auto
+
+import (
+	"context"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	git "gopkg.in/src-d/go-git.v4"
+	"gopkg.in/src-d/go-git.v4/plumbing"
+)
+
+func setupGitRepo(ctx context.Context, workDir string, repoArgs *GitRepo) (string, error) {
```

Comment:
Do you have a link for reference? Let's open an issue here. My assumption was that this could be handled via setting environment variables but making the token explicit makes sense and is more well-scoped.